pkg/adaptor/hypervisor/vsphere/service.go

```diff
@@ -145,6 +145,12 @@ func (s *hypervisorService) CreateVM(ctx context.Context, req *pb.CreateVMReques
 
 func (s *hypervisorService) StartVM(ctx context.Context, req *pb.StartVMRequest) (*pb.StartVMResponse, error) {
 
+	sess, lerr := VcenterLogin(s.vim25Client, s.serviceConfig)
```

Comment:
Hmm do you really need login/logout for every API call? I feel it'll slow down all the operator and don't really see any use for it.
Can't the login/logout be tied to server start and shutdown (in server.go) ?

Reply:
Thats the way it was originally. Unfortunately the login session can timeout in between a create and destroy VM which produces an error. This is a stop gap hack until I figure out how the session manager works and come up with something better. I don't see how it can slow down the operator much as how often do we expect VMs to be created and destroyed?

Reply:
Plus logging in is a fraction of the total time it takes to create the VM. Cloning the template is the long pole.